onmt/inputters/text_utils.py

```diff
@@ -81,12 +111,22 @@ def numericalize(vocabs, example):
                 f"Something went wrong with task {vocabs['data_task']}"
         )
 
+    # Numericalize src feats
     if 'src_feats' in vocabs.keys():
-        for featname in vocabs['src_feats'].keys():
-            src_feat = example['src'][featname].split()
-            vf = vocabs['src_feats'][featname]
-            # we'll need to change this if we introduce tgt feat
-            numeric['src'][featname] = vf(src_feat)
+        numeric_feats = []
+        for feat_vocab, feat in zip(
+                vocabs['src_feats'], example['src']['feats']):
+            numeric_feats.append(feat_vocab(feat.split()))
+        numeric['src']['feats'] = numeric_feats
+
```

Comment:
ok here it gets tricky.
don't you think we need to use numeric['src']['feats_ids'] for clarity ?